### DISCUSSION TITLE
How can I use the GPU?

### DISCUSSION THREAD

**Discussion by Caio-Giulio-Cesare** [weecology/DeepForest#481] [Q&A]
Hi everyone.
I have been using deepforest for some time and, thanks also to your precious help, I am able to make predictions and train a new model using a PC with the classic CPU... and everything seems to work well .

Now, to make everything faster, I would like to try using the GPU.

For this purpose I have an ACER Nitro 5 Win 11 notebook - with NVIDIA GeForce RTX 2060 graphics card (on PCI Slot 1 (PCI Bus 1, device 0, function 0)).

I performed the installation with GPU as indicated in the relevant chapter (conda/mamba GPU with cudatoolkit=10.2)

I tried to set the parameters of the deepforest_config.yml file by doing various tests for example
devices: 1
or devices: auto
accelerator: gpu

but always with negative outcome, getting the error:

'No supported gpu backend found!' when I create the model (model = main.deepforest())

I also tried with
devices: auto
accelerator: auto
and then after creating the model I try to move to cuda with

model.to("cuda")

but I get the error
('Torch not compiled with CUDA enabled',)

Does anyone kindly have some indications and can kindly help me understand how I should set the various parameters to use the GPU I have available?

Thanks so much in advance to anyone who wants to help me.
Bye.

**Comment by ethanwhite** [comment]
Can you try reinstalling pytorch. There are slightly different versions for with/without GPU support.

Using pip do:

```
pip3 install torch torchvision torchaudio
```

Or using conda do:

```
conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
```

**Comment by bw4sz** [comment]
Agreed. Before you get involved in DeepForest can you confirm. Don't work on anything downstream of torch until this passes.

```
(DeepTreeAttention) [b.weinstein@c39a-s21 ~]$ python
Python 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:34:09) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.cuda.is_available()
True
```

**Reply by xabierr** [reply]
Used conda as per above to install pytorch but still getting a False return. I am using RTX 4090 GPU.

**Reply by bw4sz** [reply]
This is not a DeepForest issue, but happy to try to help. Can you create a clean env and paste the entire install process below. I'd need to see everything to guess at a next step.

**Comment by Caio-Giulio-Cesare** [comment]
Thank you both so much, you are always very kind!

**Comment by xabierr** [comment]
Thanks! Attached the entire installation process.

Running Windows 10 with a NVIDIA GeoForce RTX 4090.

Cheers,

Javier
